checkstyle/checkstyle.xml

```diff
@@ -0,0 +1,108 @@
+        <!-- Checks that the order of modifiers conforms to the suggestions in the java specification -->
+        <!-- See http://checkstyle.sourceforge.net/config_modifier.html#ModifierOrder -->
+        <module name="ModifierOrder"/>
+        <!-- Checks for braces around code blocks -->
+        <!-- See http://checkstyle.sourceforge.net/config_blocks.html#NeedBraces -->
+        <module name="NeedBraces"/>
+        <!-- Checks that there is no whitespace after a token -->
+        <!-- See http://checkstyle.sourceforge.net/config_whitespace.html#NoWhitespaceAfter -->
+        <module name="NoWhitespaceAfter">
+            <property name="tokens" value="ARRAY_INIT"/>
+            <property name="allowLineBreaks" value="true"/>
+        </module>
+        <module name="NoWhitespaceAfter">
+            <property name="tokens" value="BNOT, DEC, DOT, INC, LNOT, UNARY_MINUS, UNARY_PLUS"/>
+            <property name="allowLineBreaks" value="false"/>
+        </module>
+        <!-- Checks that there is no whitespace before a token -->
+        <!-- See http://checkstyle.sourceforge.net/config_whitespace.html#NoWhitespaceBefore -->
+        <module name="NoWhitespaceBefore">
+            <property name="tokens" value="DOT"/>
+            <property name="allowLineBreaks" value="true"/>
+        </module>
+        <module name="NoWhitespaceBefore">
+            <property name="tokens" value="SEMI, POST_DEC, POST_INC"/>
+            <property name="allowLineBreaks" value="false"/>
+        </module>
+        <!-- Checks for redundant import statements -->
+        <!-- See http://checkstyle.sourceforge.net/config_imports.html#RedundantImport -->
+        <module name="RedundantImport"/>
+        <!-- Checks the placement of right curly braces -->
+        <!-- See http://checkstyle.sourceforge.net/config_blocks.html#RightCurly -->
+        <module name="RightCurly"/>
+        <!-- Checks that string literals are not used with == or != -->
+        <!-- See http://checkstyle.sourceforge.net/config_coding.html#StringLiteralEquality -->
+        <module name="StringLiteralEquality"/>
+        <!-- Checks for unused import statements -->
+        <!-- See http://checkstyle.sourceforge.net/config_imports.html#UnusedImports -->
+        <module name="UnusedImports"/>
+        <!-- Checks that long constants are defined with an upper ell -->
+        <!-- See http://checkstyle.sourceforge.net/config_misc.html#UpperEll -->
+        <module name="UpperEll"/>
+        <!-- Checks that a token is followed by whitespace -->
+        <!-- See http://checkstyle.sourceforge.net/config_whitespace.html#WhitespaceAfter -->
+        <module name="WhitespaceAfter"/>
+        <!-- Checks that a token is surrounded by whitespace -->
+        <!-- See http://checkstyle.sourceforge.net/config_whitespace.html#WhitespaceAround -->
+        <module name="WhitespaceAround"/>
+    </module>
+    <!-- See http://checkstyle.sourceforge.net/config_imports.html#FileTabCharacter -->
```

Comment:
This is not the right URL, should be `http://checkstyle.sourceforge.net/config_whitespace.html#FileTabCharacter`